dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/csv/DefaultCsvImportService.java

```diff
@@ -365,6 +373,37 @@ private List<CategoryOption> categoryOptionsFromCsv( CsvReader reader )
         return list;
     }
 
+    private List<Indicator> indicatorsFromCsv( CsvReader reader )
+        throws IOException
+    {
+        List<Indicator> list = new ArrayList<>();
+        Map<String, IndicatorType> indicatorTypeMap = new HashMap<>();
+
+        while ( reader.readRecord() )
+        {
+            String[] values = reader.getValues();
+
+            if ( values != null && values.length > 0 )
+            {
+                Indicator object = new Indicator();
+                setIdentifiableObject( object, values );
+                object.setDescription( getSafe( values, 3, 255 ) );
+                object.setShortName( getSafe( values, 4, object.getName(), 50 ) );
+                object.setDenominator( getSafe( values, 5 ) );
+                object.setDenominatorDescription( getSafe( values, 6, 255 ) );
+                object.setNumerator( getSafe( values, 7 ) );
+                object.setNumeratorDescription( getSafe( values, 8, 255 ) );
+                object.setAnnualized( Boolean.parseBoolean( getSafe( values, 9, Boolean.FALSE.toString(), 40 ) ) );
+                object.setDecimals( NumberUtils.toInt( getSafe( values, 10 ), 0 ) );
+                object.setIndicatorType( indicatorTypeMap.computeIfAbsent( getSafe( values, 11 ),
+                    key -> manager.get( IndicatorType.class, key ) ) );
```

Comment:
This can result in `null` without causing an error. Not sure that is the best behaviour. I'd rather have this throw an exception if a indicator type is used that does not exist.